From bee3a63142722dcc42cfa61d9a53810f2c6be900 Mon Sep 17 00:00:00 2001
From: Craig Barratt <19445341+craigbarratt@users.noreply.github.com>
Date: Sun, 26 Nov 2023 16:39:49 -0800
Subject: [PATCH 19/20] minor tweak

---
 main.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/main.c b/main.c
index d3add21..1ad8cee 100644
--- a/main.c
+++ b/main.c
@@ -290,11 +290,11 @@ static void output_itemized_counts(const char *prefix, int *counts)
 		counts[0] -= counts[1] + counts[2] + counts[3] + counts[4];
 		for (j = 0; j < 5; j++) {
 			if (counts[j]) {
-				tmplen = snprintf(buf+len, sizeof buf - len - 2,
+				tmplen = snprintf(buf+len, sizeof(buf) - len - 2,
 					"%s%s: %s",
 					pre, labels[j], comma_num(counts[j]));
 				// did snprintf overflow?
-				if (tmplen >= (sizeof buf - len - 2)) {
+				if (tmplen >= (int)sizeof(buf) - len - 2) {
 					break;
 				}
 				len += tmplen;
-- 
2.49.0

