From 1d55988cd2fd020e3b0b6c353bcb53ce7ca7f23e Mon Sep 17 00:00:00 2001
From: Craig Barratt <19445341+craigbarratt@users.noreply.github.com>
Date: Sun, 26 Nov 2023 16:12:39 -0800
Subject: [PATCH 18/20] minor formatting

---
 backuppc/bpc_poolWrite.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/backuppc/bpc_poolWrite.c b/backuppc/bpc_poolWrite.c
index 4f411fb..17dbbe5 100644
--- a/backuppc/bpc_poolWrite.c
+++ b/backuppc/bpc_poolWrite.c
@@ -795,12 +795,12 @@ void bpc_poolWrite_addToPool(bpc_poolWrite_info *info, char *fileName, int v3Poo
          * If rename and regular file copy failed
          * then try again.
          */
-            if( unlink(poolPath->s) ) {
+            if ( unlink(poolPath->s) ) {
             	bpc_logErrf("bpc_poolWrite_addToPool: remove %s failed; errno = %d\n", poolPath->s, errno);
             	bpc_unlockRangeFile(lockFd);
             	unlink(lockFile->s);
 		redo = 1;
-            } else if( rename(fileName, poolPath->s) ) {
+            } else if ( rename(fileName, poolPath->s) ) {
             	bpc_logErrf("bpc_poolWrite_addToPool: rename %s -> %s failed; errno = %d\n", fileName, poolPath->s, errno);
             	bpc_unlockRangeFile(lockFd);
             	unlink(lockFile->s);
@@ -811,7 +811,7 @@ void bpc_poolWrite_addToPool(bpc_poolWrite_info *info, char *fileName, int v3Poo
 		}
 	    }
 
-	    if(!redo) {
+	    if ( !redo ) {
             	chmod(poolPath->s, 0444);
             	stat(poolPath->s, &st);
             	info->retValue     = v3PoolFile ? 2 : 0;
-- 
2.49.0

